### PR TITLE
Error improvements 2

### PR DESCRIPTION
**Pull request by avitova**
This PR should contain some of the 'less questionable' error refactoring. More or less, this is just coming up with user-facing error messages.
This also removes the Dao Initialization error since we no more return errors for the initialization.

### CODE REVIEW

File: internal/payloads/error_payload.go [hunk 1]
```diff
@@ -108,27 +106,13 @@ func NewNotFoundError(ctx context.Context, err error) *ResponseError {
...
-		logger.Error().Msg(msg)
-	}
-	return &ResponseError{
-		HTTPStatusCode: 500,
-		Message:        msg,
-		TraceId:        ctxval.TraceId(ctx),
-		Error:          err,
-	}
-}
-
```
Comment (adiabramovitch) [inline internal/payloads/error_payload.go:123]: don't we need that error?

File: internal/services/instance_types_service.go [hunk 2]
```diff
@@ -35,18 +35,18 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
...
 	if err != nil {
-		renderError(w, r, payloads.NewAWSError(r.Context(), "failed to establish ec2 connection", err))
+		renderError(w, r, payloads.NewAWSError(r.Context(), "unable to get AWS client", err))
 		return
 	}
 
 	instances, err := ec2Client.ListInstanceTypesWithPaginator(r.Context())
 	if err != nil {
-		renderError(w, r, payloads.NewAWSError(r.Context(), "can't list EC2 instance types", err))
+		renderError(w, r, payloads.NewAWSError(r.Context(), "unable to list AWS instances", err))
```
Comment (adiabramovitch) [inline internal/services/instance_types_service.go:44]: I would add EC2 as well but it is just a nit

File: internal/services/instance_types_service.go [hunk 3]
```diff
@@ -14,7 +14,7 @@ func ListInstanceTypes(w http.ResponseWriter, r *http.Request) {
 	sourceId := chi.URLParam(r, "ID")
 	region := r.URL.Query().Get("region")
 	if region == "" {
-		renderError(w, r, payloads.NewMissingRequestParameterError(r.Context(), "region"))
+		renderError(w, r, payloads.NewMissingRequestParameterError(r.Context(), "region parameter missing"))
```
Comment (adiabramovitch) [inline internal/services/instance_types_service.go:17]: region parameter is missing

**Review by adiabramovitch — APPROVED** [review]
LGTM, just small nits

File: internal/payloads/error_payload.go (hunk 1 above)
Comment (lzap) [reply internal/payloads/error_payload.go:123]: Yep, that is correct. No DAO init errors, yay!!! :-)

Comment (adiabramovitch) [reply internal/payloads/error_payload.go:123]: great! removing it is :)

**Review by lzap — APPROVED** [review]
I am good thanks, leaving the rest on @adiabramovitch

**Comment by adiabramovitch** [comment]
@avitova Thanks!

**Pull request merged**